### PR TITLE
feat(github): refactor forkMode to reduce API queries

### PR DESCRIPTION
**Pull request by rarkins**
## Changes

Refactors fork mode to query repo's list of forks instead of forker's list of repos.

## Context

This should be product the same result as before.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/platform/github/index.ts [hunk 1]
```diff
@@ -241,6 +242,87 @@ export async function getJsonFile(
...
+): Promise<GhRestRepo | null> {
+  const forks = await listForks(token, repository);
+  const orgs = await getForkOrgs(token);
+  if (!orgs.length) {
+    throw new Error(REPOSITORY_CANNOT_FORK);
+  }
+  let forkedRepo: GhRestRepo | undefined;
+  for (const forkOrg of orgs) {
+    logger.debug(`Searching for forked repo in ${forkOrg}`);
+    forkedRepo = forks.find((repo) => repo.owner.login === forkOrg);
```
Comment (viceice) [inline lib/modules/platform/github/index.ts:289]: forks can be a huge list, so better use a map where key is the owner login.

File: lib/modules/platform/github/index.ts [hunk 2]
```diff
@@ -241,6 +242,87 @@ export async function getJsonFile(
...
+  if (!config.renovateForkUser) {
+    try {
+      logger.debug('Determining fork user from API');
+      const userDetails = await getUserDetails(platformConfig.endpoint, token);
+      config.renovateForkUser = userDetails.username;
+    } catch (err) {
+      logger.debug({ err }, 'Error getting username for forkToken');
+    }
+  }
+  return config.renovateForkUser ? [config.renovateForkUser] : [];
```
Comment (viceice) [inline lib/modules/platform/github/index.ts:256]: do you want to support multiple fork orgs later?

Comment (rarkins) [reply lib/modules/platform/github/index.ts:256]: Yes, see next PR. This OR was derived from it

File: lib/modules/platform/github/index.ts (hunk 1 above)
Comment (rarkins) [reply lib/modules/platform/github/index.ts:289]: Do you think it really takes that much time to iterate through?

Comment (viceice) [reply lib/modules/platform/github/index.ts:289]: depends a little bit of the length of the outer loop. for a single org it's no problem, but for 3 org you iterate three time in worst case. so its getting more worse with more orgs

Comment (rarkins) [reply lib/modules/platform/github/index.ts:289]: It should be at most 2 orgs. Also, I can't easily replicate more than 1 org in this PR, so I'd like to push this to the next PR if necessary

**Review by viceice — APPROVED** [review]

**Pull request merged**